indexer/base/kind.go

```diff
@@ -41,14 +47,14 @@ const (
 	Uint64Kind
 
 	// IntegerKind represents an arbitrary precision integer number. Values of this type must
-	// be of the go type int64, string or a type that implements fmt.Stringer with the resulted string
-	// formatted as an integer number.
+	// be of the go type string and formatted as base10 integers, specifically matching to
+	// the IntegerFormat regex.
 	IntegerKind
```

Comment:
Would it be better to specify go `*big.Int` here for simpler validation?

Reply:
Or should we at least have some limit on the number of digits for this and also for `Decimal`? It's probably better to be specific to avoid overflow conditions.

Postgres says `NUMERIC` can handle:
> up to 131072 digits before the decimal point; up to 16383 digits after the decimal point
But that is probably WAY more than any realistic use case we'll see.

Reply:
i think we should set some upper bound

Reply:
How about 100 digits max for integers and 50 digits either side of the decimal point (100 total) with +/-99 exponent? Seems like Postgres can handle that fine. Not sure about other systems.

Reply:
I just added the above restrictions. Let me know if anyone objects